### ISSUE TITLE
Add stunnel-style forwarding from TCP -> TLS

### ISSUE THREAD

**Issue by cfcs** [hannesm/tlstunnel#29]
Would be nice for wrapping local outgoing connections.

**Comment by hannesm** [comment]
you mean listening locally on a TCP port, and forwarding all connections via tls to a remote host?

**Comment by cfcs** [comment]
yes, exactly that (a "proxy" as opposed to a "reverse proxy", I guess)
